src/java/com/zimbra/cs/nginx/NginxLookupExtension.java

```diff
@@ -421,6 +424,11 @@ private NginxLookupRequest checkRequest(HttpServletRequest httpReq) throws Nginx
                 req.isZimbraAdmin = Boolean.parseBoolean (isZimbraAdmin);
             }
 
+            String isZimbraZx = httpReq.getHeader(AUTH_ZIMBRA_ZX);
```

Comment:
Lets have this as else if.

Reply:
I've switched to an enum here, the admin has priority over zx since it's the last one and both can't happen anymore